cpp/src/io/utilities/config_utils.cpp

```diff
@@ -87,38 +89,160 @@ bool is_stable_enabled() { return is_all_enabled() or get_env_policy() == usage_
 
 }  // namespace nvcomp_integration
 
-inline std::mutex& host_mr_lock()
+}  // namespace detail
+
+namespace {
+class fixed_pinned_pool_memory_resource {
+  using upstream_mr    = rmm::mr::pinned_host_memory_resource;
+  using host_pooled_mr = rmm::mr::pool_memory_resource<upstream_mr>;
+
+ private:
+  upstream_mr upstream_mr_{};
+  size_t pool_size_{0};
+  // Raw pointer to avoid a segfault when the pool is destroyed on exit
+  host_pooled_mr* pool_{nullptr};
+  void* pool_begin_{nullptr};
+  void* pool_end_{nullptr};
+  cuda::stream_ref stream_{cudf::detail::global_cuda_stream_pool().get_stream(0).value()};
```

Comment:
Do we always have at least one stream in `global_cuda_stream_pool()`?

Reply:
We do. Calls to `fork_streams` would fail otherwise.
But now that I look at the stream pool implementation, I think I should use `get_stream()` without the index. With this, we get the "next" stream from the pool, rather than the first one. Will change this, unless you disagree.